src/app/wallet/proposals/add-proposal/add-proposal.component.ts

```diff
@@ -0,0 +1,151 @@
+import { SnackbarService } from 'app/core/snackbar/snackbar.service';
+import { Profile } from 'app/core/market/api/profile/profile.model';
+import { PeerService } from 'app/core/rpc/peer/peer.service';
+
+import {
+  ProposalConfirmationComponent
+} from 'app/modals/proposal-confirmation/proposal-confirmation.component';
+import { ModalsHelperService } from 'app/modals/modals-helper.service';
+import { Proposal } from 'app/wallet/proposals/models/proposal.model';
+
+@Component({
+  selector: 'app-add-proposal',
+  templateUrl: './add-proposal.component.html',
+  styleUrls: ['./add-proposal.component.scss']
+})
+export class AddProposalComponent implements OnInit {
+
+  log: any = Log.create('add-item.component');
+  private destroyed: boolean = false;
+  public isTnCAccepted: boolean = false;
+  private profileId: number;
+  // form controls
+  public proposalFormGroup: FormGroup;
+
+  constructor(
+    private router: Router,
+    private formBuilder: FormBuilder,
+    private dialog: MatDialog,
+    private modalService: ModalsHelperService,
+    private peerService: PeerService,
+    private proposalsService: ProposalsService,
+    private profileService: ProfileService,
+    private snackbarService: SnackbarService
+  ) { }
+
+  ngOnInit() {
+    // get default profile address.
+    this.profileService.default().takeWhile(() => true).subscribe((profile: Profile) => {
+      this.profileId = profile.id;
+    })
+
+    this.proposalFormGroup = this.formBuilder.group({
+      title: ['', Validators.compose([Validators.required, Validators.maxLength(50)])],
+      description: ['', Validators.compose([Validators.required, Validators.maxLength(2000)])],
+      options: this.formBuilder.array([
+        this.initOptionFields(),
+        this.initOptionFields()
+      ]),
+
+      // @TODO `nickname` and `email` use in the "Contact Information" section.
+      // nickname: ['', Validators.compose([Validators.maxLength(50)])],
+      // email: ['',  Validators.compose([Validators.maxLength(150)])]
+    });
+
+  }
+
+  initOptionFields(): FormGroup {
+    return this.formBuilder.group({
+      option: ['', Validators.compose([Validators.required, Validators.maxLength(50)])]
+    });
+  }
+
+  /**
+   * Programmatically generates a new option input field
+   *
+   * @public
+   * @method addNewInputField
+   * @return {none}
+   */
+  addNewInputField(): void {
+    const control = <FormArray>this.proposalFormGroup.controls.options;
+    control.push(this.initOptionFields());
+  }
+
+
+  /**
+   * Programmatically removes a recently generated option input field
+   *
+   * @public
+   * @method removeInputField
+   * @param i    {number}      The position of the object in the array that needs to removed
+   * @return {none}
+   */
+  removeInputField(i: number): void {
+    const control = <FormArray>this.proposalFormGroup.controls.options;
+    control.removeAt(i);
+  }
+
+  cancelAndDiscard() {
```

Comment:
The actions in this method are not exactly cancelling or discarding, but seem to primarily be navigating. The method should either be named something different, or alternatively, the code that makes up its body should be moved to a separate function that the cancelAndDiscard() method then invokes.

Reply:
Yes, I agree, @zaSmilingIdiot. :)

I changed it as backToProposals()
bcz, currently, are doing nothing in the cancelAndDiscard(), just navigate to the proposals page by the same.